### DISCUSSION TITLE
Convert to / from Java BigDecimal

### DISCUSSION THREAD

**Discussion by dankito** [ionspin/kotlin-multiplatform-bignum#216] [Ideas]
Hi,

I created some extension methods to convert BigDecimal to / from Java BigDecimal (see below).

Two questions here:

- Are they correct and do you have any suggestions how to improve them (especially for converting from)?

- If i give you a pull request with them, would you like to integrate it into the jvmMain branch of your library?

```
package net.dankito.banking.extensions

import com.ionspin.kotlin.bignum.integer.Sign
import java.math.BigDecimal


fun BigDecimal.toIonspinBigDecimal(): com.ionspin.kotlin.bignum.decimal.BigDecimal {
    return com.ionspin.kotlin.bignum.decimal.BigDecimal.parseString(this.toString()) // TODO: couldn't figure out how to create BigDecimal from unscaledValue and scale
}

fun com.ionspin.kotlin.bignum.decimal.BigDecimal.toJavaBigDecimal(): BigDecimal {
    val converted = BigDecimal.valueOf(this.significand.longValue(), (this.precision - this.exponent - 1).toInt())

    if (this.significand.sign == Sign.NEGATIVE) {
        return converted.negate()
    }

    return converted
}
```

**Comment by ionspin** [comment]
Hi @dankito , if you find these converters useful, I'd love to add them to the library, so a pull request would be great. 

As far as the correctness goes, conversion from java big decimal using string as intermediary would work, but I think it would be too inefficient, so I'd suggest solving that with a more direct approach (i.e. by using Java BigDecimal internal Java BigInteger intValue and scale)

For the conversion to Java Big Decimal, the result wouldn't be completely correct, because `significand` can be larger than a long, so just calling `longValue()` would lose information. There's already some conversion done in the test classes, so you might take some code from there as inspiration (https://github.com/ionspin/kotlin-multiplatform-bignum/blob/master/bignum/src/jvmTest/kotlin/com/ionspin/kotlin/bignum/decimal/DecimalUtility.kt)

**Reply by BraisGabin** [reply]
Could that Utility file be moved to the main library? I know, it would be java only available for java but it would help with interoperability. I was going to ask to implement `Number` but I saw #170.

**Reply by ionspin** [reply]
Hi @BraisGabin I would be absolutely fine with that, unfortunately currently I don't have a lot of time, so if you could add them in a pull request that would be great!

**Reply by ionspin** [reply]
Oh and another reason I am not going to implement Number is here #208
